test/unit/test_logger.py

```diff
@@ -126,3 +126,25 @@ def test_cyan_text():
     x = '{}{}{}'.format(colorama.Fore.CYAN, 'foo', colorama.Style.RESET_ALL)
 
     assert x == logger.cyan_text('foo')
+
+
+def test_markup_detection_pycolors0(monkeypatch):
+    monkeypatch.setenv('PY_COLORS', '0')
+    assert not logger.should_do_markup()
+
+
+def test_markup_detection_pycolors1(monkeypatch):
+    monkeypatch.setenv('PY_COLORS', '1')
+    assert logger.should_do_markup()
+
+
+def test_markup_detection_tty_yes(mocker):
+    mocker.patch('sys.stdout.isatty', return_value=True)
+    mocker.patch('os.environ', {'TERM': 'xterm'})
+    assert logger.should_do_markup()
```

Comment:
As per my previous explanation @ https://github.com/ansible/molecule/pull/1533#discussion_r260881603, if you use `pytest-mock`, you have to manually unpatch everything. Otherwise, it causes external side-effects which break at least some of pytest versions.

Reply:
```suggestion
    assert logger.should_do_markup()
    mocker.resetall()
    mocker.stopall()
```